### PR TITLE
feat: Add instrumentation for InvokeModelCommand

### LINKED ISSUE #221: Instrument InvokeModelCommand @aws-sdk/client-bedrock-runtime
Instrument BedrockRuntimeClient.prototype.send when command is InvokeModelCommand and model is not `amazon.titan-embed` to create the span `/Llm/completion/Bedrock/InvokeModel/Command` and the and the relevant LlmChatCompletionSummary and LlmChatCompletionMessage. When model is amazon.titan-embed create the span with the name of Llm/embedding/Bedrock/InvokeModelCommand and the relevant LlmEmbedding.

**NOTES**:
 * This instrumentation should only register when `ai_monitoring.enabled` and `feature_flag.aws_bedrock_instrumentation` are both true. 
 * The aws sdk v3 instrumentation is middleware based so this is not a traditional span.wrap/record effort.

### PR DESCRIPTION
**Pull request by bizob2828**
- refactor: updated init to return a value to decide to register middleware
- feat: finished instrumentation
- test: added common test assertions for chat completion
- test: updated tests to run in a loop for all models

## Description

This PR addds instrumentation for InvokeModelCommand. It creates the relevant embedding/completion spans and their respective LLM custom events.

This does not handle errors, nor streaming, or storing response id context for feedback messages.  All of that is in separate stories.

It is also worth noting the registration of middleware has changed so it returns a value. This is used in this PR to decide if the feature ai monitoring is enabled.

## How to Test

`npm run versioned`

## Related Issues

Closes #221

### CODE REVIEW

File: lib/v3/bedrock.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+  /** 🚨 Code Smell 🚨
+   * spec functions cannot be async, nor can after hooks.
+   * this works due to the nature of the event loop.
+   * the promise resolves before the after hook fires.
+   */
+  let credentials = null
+  config.credentials().then((creds) => {
+    credentials = creds
+  })
```
Comment (jsumners-nr) [inline lib/v3/bedrock.js:62]: We should definitely create a ticket to fix this.

**Review by jsumners-nr — APPROVED** [review]
Looks good to me.

File: lib/v3/bedrock.js (hunk 1 above)
Comment (bizob2828) [reply lib/v3/bedrock.js:62]: yea i can log something in the agent. it'd basically be allowing async spec retrieval and also perhaps the after hook

Comment (bizob2828) [reply lib/v3/bedrock.js:62]: https://github.com/newrelic/node-newrelic/issues/1936

File: tests/versioned/v3/bedrock-chat-completions.tap.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+'use strict'
+
+const tap = require('tap')
+const utils = require('@newrelic/test-utilities')
+utils(tap)
+const common = require('../common')
+const createAiResponseServer = require('../aws-server-stubs/ai-server')
+const { FAKE_CREDENTIALS } = require('../aws-server-stubs')
+const requests = {
```
Comment (bizob2828) [inline tests/versioned/v3/bedrock-chat-completions.tap.js:14]: i plan to flesh more out on the prompts to get all the keys in a diff PR, maybe the next one i do

**Pull request merged**